pkgs/development/libraries/libwnck/3.x.nix

```diff
@@ -1,31 +1,72 @@
-{ stdenv, fetchurl, pkgconfig, libX11, gtk3, intltool, gobject-introspection, gnome3 }:
+{ stdenv
+, fetchurl
+, fetchpatch
+, meson
+, ninja
+, pkgconfig
+, gtk-doc
+, docbook_xsl
+, docbook_xml_dtd_412
+, libX11
+, glib
+, gtk3
+, libstartup_notification
+, gettext
+, gobject-introspection
+, gnome3
+}:
 
-let
+stdenv.mkDerivation rec{
   pname = "libwnck";
-  version = "3.30.0";
-in stdenv.mkDerivation rec{
-  name = "${pname}-${version}";
+  version = "3.32.0";
+
+  outputs = [ "out" "dev" "devdoc" ];
+  outputBin = "dev";
 
   src = fetchurl {
-    url = "mirror://gnome/sources/${pname}/${stdenv.lib.versions.majorMinor version}/${name}.tar.xz";
-    sha256 = "0f9lvhm3w25046dqq8xyg7nzggxpmdriwrb661nng05a8qk0svdc";
+    url = "mirror://gnome/sources/${pname}/${stdenv.lib.versions.majorMinor version}/${pname}-${version}.tar.xz";
+    sha256 = "1jp3p1lnwnwi6fxl2rz3166cmwzwy9vqz896anpwc3wdy9f875cm";
   };
 
-  outputs = [ "out" "dev" "devdoc" ];
-  outputBin = "dev";
+  patches = [
+    # https://gitlab.gnome.org/GNOME/libwnck/issues/139
+    (fetchpatch {
+      url = https://gitlab.gnome.org/GNOME/libwnck/commit/0d9ff7db63af568feef8e8c566e249058ccfcb4e.patch;
+      sha256 = "18f78aayq9jma54v2qz3rm2clmz1cfq5bngxw8p4zba7hplyqsl9";
+    })
+    # https://gitlab.gnome.org/GNOME/libwnck/merge_requests/12
+    ./fix-pc-file.patch
+  ];
+
+  nativeBuildInputs = [
+    meson
+    ninja
+    pkgconfig
+    gettext
+    gobject-introspection
+    gtk-doc
+    docbook_xsl
+    docbook_xml_dtd_412
+  ];
 
-  configureFlags = [ "--enable-introspection" ];
+  buildInputs = [
```

Comment:
oop, really quick.

I see `pango cairo` in `Requires.private` so it probably should be here formally

Reply:
Done, thanks.